http3/server.go

```diff
@@ -213,6 +213,23 @@ func (s *Server) maxHeaderBytes() uint64 {
 }
 
 func (s *Server) handleRequest(sess quic.Session, str quic.Stream, decoder *qpack.Decoder, onFrameError func()) requestError {
+
+	var (
+		hdrDeadline      time.Time
+		wholeReqDeadline time.Time
+		writeDeadline    time.Time
+	)
+
+	t0 := time.Now()
+	if d := s.ReadHeaderTimeout; d != 0 {
```

Comment:
The documentation of `http.Server` says:
```
// ReadHeaderTimeout is the amount of time allowed to read
// request headers. The connection's read deadline is reset
// after reading the headers and the Handler can decide what
// is considered too slow for the body. If ReadHeaderTimeout
// is zero, the value of ReadTimeout is used. If both are
// zero, there is no timeout.
```

So don't you need to read the `ReadTimeout` if `ReadHeaderTimeout` is 0?